rust/bin-shared/src/tun_device_manager/windows.rs

```diff
@@ -367,47 +367,32 @@ fn set_iface_config(luid: wintun::NET_LUID_LH, mtu: u32) -> Result<()> {
         Value: unsafe { luid.Value },
     };
 
-    // Set MTU for IPv4
-    {
-        let mut row = MIB_IPINTERFACE_ROW {
-            Family: AF_INET,
-            InterfaceLuid: luid,
-            ..Default::default()
-        };
-
-        // SAFETY: TODO
-        unsafe { GetIpInterfaceEntry(&mut row) }.ok()?;
-
-        // https://stackoverflow.com/questions/54857292/setipinterfaceentry-returns-error-invalid-parameter
-        row.SitePrefixLength = 0;
+    try_set_mtu(luid, AF_INET, mtu)?;
+    try_set_mtu(luid, AF_INET6, mtu)?;
+    Ok(())
+}
 
-        // Set MTU for IPv4
-        row.NlMtu = mtu;
+fn try_set_mtu(luid: NET_LUID_LH, family: ADDRESS_FAMILY, mtu: u32) -> Result<()> {
+    let mut row = MIB_IPINTERFACE_ROW {
+        Family: family,
+        InterfaceLuid: luid,
+        ..Default::default()
+    };
 
-        // SAFETY: TODO
-        unsafe { SetIpInterfaceEntry(&mut row) }.ok()?;
+    // SAFETY: TODO
+    if unsafe { GetIpInterfaceEntry(&mut row) }.ok().is_err() {
+        tracing::warn!(?family, "Couldn't set MTU");
```

Comment:
Can we somehow check an error code that is the equivalent of "not found" and not log a warning on that?